src/apps/Agreement/DisputableApps/DisputableApps.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react'
+import {
+  DataView,
+  ContextMenu,
+  ContextMenuItem,
+  useTheme,
+  GU,
+  useLayout,
+  Help,
+  TokenBadge,
+} from '@aragon/ui'
+import { KnownAppBadge } from '../../../templates/kit'
+import InfoField from './../InfoField'
+import { AppItemsType } from './../prop-types'
+import SubtleLabel from './SubtleLabel'
+import TokenAmount from './TokenAmount'
+
+const EXPANDABLE_ROW_GAP = `${3 * GU}px`
+
+const DisputableApps = React.memo(({ items }) => {
+  const { layoutName } = useLayout()
+  const theme = useTheme()
+
+  return (
+    <DataView
+      fields={[
+        { label: 'App', priority: 1, childStart: true },
+        { label: 'Actions', align: 'left' },
+      ]}
+      entries={items}
+      renderEntry={entry => renderEntry(entry, layoutName)}
+      renderEntryExpansion={entry => renderEntryExpansion(entry, layoutName)}
+      renderEntryActions={entry => renderEntryActions(entry, theme)}
+    />
+  )
+})
+
+function renderEntry({ allowedActions }, layoutName) {
+  return [
+    <div
+      css={`
+        display flex;
+        align-items: center;
+
+        /* Height must match expansion button to align nicely */
+        ${(layoutName === 'medium' || layoutName === 'large') &&
+          'height: 32px;'}
+        
+      `}
+    >
+      <KnownAppBadge appName="voting.aragonpm.eth" label="Voting" />
+    </div>,
+    <React.Fragment>{allowedActions.join(', ')}</React.Fragment>,
+  ]
+}
+
+function renderEntryActions(entry, theme) {
+  const { entryActions } = entry
+
+  return (
+    <ContextMenu>
+      {entryActions.map(([onClick, Icon, label], index) => (
```

Comment:
Realizing we do this in the `PermissionsView` too, but what do you think about moving `Icon` to being a rendered node rather than component when using the `ContextMenu`?

Reply:
Sure! is this a readability change? or does this affect performance in some way?

Reply:
Solely for readability and consistency, as well as control—gives the user the ability to control props set on the icon / etc.

We may have a few sparse examples left, but we've generally taken the approach of using "slots" with already rendered components